src/test/java/org/jnode/fs/DataStructureAsserts.java

```diff
@@ -130,4 +132,84 @@ private static String toHexString(byte[] digest)
 
         return builder.toString();
     }
-}
+
+    /**
+     * Builds up the structure for the given file system entry to get the metadata.
+     *
+     * @param entry  the entry to process.
+     * @param actual the string to append to.
+     * @param indent the indent level.
+     * @throws IOException if an error occurs.
+     */
+    public static void buildXfsMetaDataStructure(FSEntry entry, StringBuilder actual, String indent) throws IOException
```

Comment:
The XFS specific tests should be either in a separate class in the `org.jnode.fs.xfs` test folder, maybe even incorporated into the `XfsFileSystemTest` class itself. The contents of this class should stay generic, or make sense for all the filesystem types supported.

Reply:
The class was changed. The metadata functions were defined in the XfsFileSystemTest.